File: 128. Longest Consecutive Sequence/main.py
from typing import List

class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        result = 0
        new_set = set(nums)
        # Loop through the set to ignore duplicated items like this [1,1,1,1,1,2,3,4]
        for num in new_set:
            if num-1 not in new_set:
                current_sum = 1
                curr_num = num
                while curr_num + 1 in new_set:
                    current_sum += 1
                    curr_num+=1
                result = max(result, current_sum)
        return result

if __name__ == "__main__":
    s = Solution()
    print(s.longestConsecutive([0,3,7,2,5,8,4,6,0,1]))

File: 128. Longest Consecutive Sequence/main.js
/**
 * @param {number[]} nums
 * @return {number}
 */
const longestConsecutive = function (nums) {
  let result = 0;
  const new_set = new Set(nums);

  for (const num of new_set) {
    if (new_set.has(num - 1)) {
      continue;
    }

    let curr_num = num;
    let curr_sum = 1;

    while (new_set.has(curr_num + 1)) {
      curr_num += 1;
      curr_sum += 1;
    }
    result = Math.max(result, curr_sum);
  }

  return result;
};
